src/core/ext/filters/client_channel/subchannel.h

```diff
@@ -85,9 +85,11 @@ class ConnectedSubchannel : public RefCountedWithTracing<ConnectedSubchannel> {
     size_t parent_data_size;
   };
 
-  explicit ConnectedSubchannel(grpc_channel_stack* channel_stack,
-                               channelz::SubchannelNode* channelz_subchannel,
-                               intptr_t socket_uuid);
+  explicit ConnectedSubchannel(
+      grpc_channel_stack* channel_stack,
+      grpc_core::RefCountedPtr<grpc_core::channelz::SubchannelNode>
```

Comment:
I don't think it's necessary for the connected subchannel to take a ref to the channelz node anymore.

Reply:
Reverted

Reply:
Actually, reverting this causes ASAN failures to show up again:

```
=================================================================
==635==ERROR: AddressSanitizer: heap-use-after-free on address 0x60e0000075b0 at pc 0x00000063cad2 bp 0x7ffddc050ac0 sp 0x7ffddc050ab8
READ of size 8 at 0x60e0000075b0 thread T0
    #0 0x63cad1 in grpc_core::channelz::CallCountingHelper::RecordCallFailed() /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/channel/channelz.cc:81:3
    #1 0x62098c in grpc_core::channelz::SubchannelNode::RecordCallFailed() /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/ext/filters/client_channel/client_channel_channelz.h:83:43
    #2 0x61f0fa in recv_trailing_metadata_ready(void*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/ext/filters/client_channel/subchannel.cc:782:26
    #3 0x5347aa in exec_ctx_run(grpc_closure*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/exec_ctx.cc:40:3
    #4 0x542cc1 in grpc_closure_run(char const*, int, grpc_closure*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/iomgr/closure.h:258:5
    #5 0x84dbea in recv_trailing_metadata_ready(void*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/ext/filters/message_size/message_size_filter.cc:185:3
    #6 0x5347aa in exec_ctx_run(grpc_closure*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/exec_ctx.cc:40:3
    #7 0x542cc1 in grpc_closure_run(char const*, int, grpc_closure*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/iomgr/closure.h:258:5
    #8 0x8a30e4 in recv_trailing_metadata_ready(void*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/ext/filters/http/client/http_client_filter.cc:191:3
    #9 0x5347aa in exec_ctx_run(grpc_closure*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/exec_ctx.cc:40:3
    #10 0x534314 in grpc_core::ExecCtx::Flush() /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/exec_ctx.cc:134:9
    #11 0x51e89a in grpc_core::ExecCtx::~ExecCtx() /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/iomgr/exec_ctx.h:108:5
    #12 0x586b55 in grpc_call_cancel /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/surface/call.cc:588:1
    #13 0x51576f in main /usr/local/google/home/ncteisen/Desktop/grpc/test/core/end2end/goaway_server_test.cc:332:3
    #14 0x7feb39ea32b0 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x202b0)
    #15 0x41ff99 in _start (/usr/local/google/home/ncteisen/Desktop/grpc/bins/asan/goaway_server_test+0x41ff99)

0x60e0000075b0 is located 48 bytes inside of 160-byte region [0x60e000007580,0x60e000007620)
freed by thread T3 (grpc_global_tim) here:
    #0 0x4d5428 in __interceptor_free.localalias.0 (/usr/local/google/home/ncteisen/Desktop/grpc/bins/asan/goaway_server_test+0x4d5428)
    #1 0x6baf9e in gpr_free /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/gpr/alloc.cc:77:3
    #2 0x59f766 in void grpc_core::Delete<grpc_core::channelz::BaseNode>(grpc_core::channelz::BaseNode*) /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/gprpp/memory.h:63:5
    #3 0x59f66d in grpc_core::RefCounted<grpc_core::channelz::BaseNode>::Unref() /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/gprpp/ref_counted.h:57:7
    #4 0x6204a1 in grpc_core::RefCountedPtr<grpc_core::channelz::SubchannelNode>::reset() /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/gprpp/ref_counted_ptr.h:112:36
    #5 0x61707e in subchannel_destroy(void*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/ext/filters/client_channel/subchannel.cc:198:28
    #6 0x51eece in closure_impl::closure_wrapper(void*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/./src/core/lib/iomgr/closure.h:150:3
    #7 0x5347aa in exec_ctx_run(grpc_closure*, grpc_error*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/exec_ctx.cc:40:3
    #8 0x534314 in grpc_core::ExecCtx::Flush() /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/exec_ctx.cc:134:9
    #9 0x56a280 in run_some_timers() /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/timer_manager.cc:131:30
    #10 0x569d84 in timer_main_loop() /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/timer_manager.cc:232:9
    #11 0x569b37 in timer_thread(void*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/iomgr/timer_manager.cc:279:3
    #12 0x6d1331 in grpc_core::(anonymous namespace)::ThreadInternalsPosix::ThreadInternalsPosix(char const*, void (*)(void*), void*, bool*)::{lambda(void*)#1}::operator()(void*) const /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/gprpp/thd_posix.cc:100:27
    #13 0x6d0d65 in grpc_core::(anonymous namespace)::ThreadInternalsPosix::ThreadInternalsPosix(char const*, void (*)(void*), void*, bool*)::{lambda(void*)#1}::__invoke(void*) /usr/local/google/home/ncteisen/Desktop/grpc/src/core/lib/gprpp/thd_posix.cc:74:25
    #14 0x4e3092 in __asan::AsanThread::ThreadStart(unsigned long, __sanitizer::atomic_uintptr_t*) (/usr/local/google/home/ncteisen/Desktop/grpc/bins/asan/goaway_server_test+0x4e3092)
```

Reply:
I will un-revert for now, let me know if you have thoughts on a cleaner fix

Reply:
Interesting.  I guess upon channel shutdown, the `ConnectedSubchannel` object may outlive the `grpc_subchannel` object.  This is probably fine.